Talha's PR

yotalha opened the pull request:
made an Item class that replaces int data
obj with id name and price.

Mimi848 commented on CustomeDB/CustomeDB.cpp:
```diff
@@ -4,19 +4,28 @@
+    void addAtEnd(int id, string name, int price) {
 
         Node* new_node = new Node();
-        new_node->data = val;
```
the implementation of linked list should'nt change just because ur using object,
the data copy and deletion should be handled by class for applying OOP concepts.

yotalha replied:
fixed.

Mimi848 commented on CustomeDB/CustomeDB.cpp:
```diff
@@ -65,55 +74,154 @@ class LinkedList {
+    l1.displayList();
+    l1.sortByNameDec();
     l1.displayList();
 
```
there is memory leakage in your program. i want u to find it out. for hint run add data and delete at head hundreds of thousands of time try testing it.

yotalha replied:
fixed.

Mimi848 reviewed (changes requested):
for(int i = 0; i < 10000000 ; i++){
        l1.addAtEnd(1, "talha", 20);
        l1.addAtEnd(2, "haseed", 10);
        l1.addAtEnd(3, "bats", 25);
        l1.deleteAtHead();
        l1.deleteAtHead();
        l1.deleteAtHead();
        
    }
test case

Mimi848 reviewed (commented):
Yeah better now good work!

yotalha commented:
> Yeah better now good work!

wus up

haseebuchiha commented on CustomeDB/LinkedList.h:
```diff
@@ -0,0 +1,235 @@
+            cout << "No such File. \n";
+        }
+        else {
+            cout << "i was here" << endl;
```
Why??

yotalha replied:
cuz i was there

haseebuchiha commented on CustomeDB/Node.cpp:
```diff
@@ -0,0 +1 @@
+#include "Node.h"
```
Should not be empty

haseebuchiha commented on CustomeDB/LinkedList.cpp:
```diff
@@ -0,0 +1 @@
+#include "LinkedList.h"
```
Should not be empty

yotalha committed:
Made Item class private

made item class data private and accessing them through getters and setters.

yotalha committed:
Made Item class private

made item class data private and accessing them through getters and setters.

haseebuchiha commented on CustomeDB/Item.h:
```diff
@@ -5,9 +5,18 @@ using namespace std;
 
 class Item {
-public:
+
```
The task was to make the object data private not the name price and id

yotalha committed:
made Node data private

made node data private, made all the getter and setters in node class.

yotalha committed:
made Node data private

made node data private, made all the getter and setters in node class.

yotalha committed:
changed updatePrice func

added new parameters, removed old_val and new_val and added u_id, u_name and new_price.

yotalha committed:
changed updatePrice func

added new parameters, removed old_val and new_val and added u_id, u_name and new_price.

yotalha committed:
Made employee class

made employee class with private members that hav getters and setters.

yotalha committed:
Merge pull request #12 from C-Training/employee

Made employee class

hamzafj5 committed:
Class client made (#14)

TalhaAsjad committed:
Made project class and getter setters

TalhaAsjad committed:
Merge pull request #15 from C-Training/ProjectClass

Made project class and getter setters

yotalha committed:
Minor refactir

yotalha committed:
commented out test class

commented test.cpp and test.h so everyone tasks can run without error.

yotalha committed:
employee Node and LinkedList

Update Node class and LinkedList class according to Employee class also updated all functions(overloading functions.)

yotalha committed:
Merge pull request #16 from C-Training/employee-node-linkedlist

employee Node and LinkedList

yotalha committed:
Activity class template

hamzafj5 committed:
Did everything again with changes in node and LL. (#18)

hamzafj5 committed:
Made the acitivity plus changed the names of my functions. (#20)

haseebuchiha committed:
Revert "Made the acitivity plus changed the names of my functions. (#20)" (#21)

This reverts commit c0f99fef92b8c3abb8864d742df7b51fb2fd1675.

yotalha committed:
ggez
